terraform/nomad-whisper/create-nomad-whisper-instance.sh

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env bash
+set -x
+unset SSH_USER
+
+#IF THE CURRENT DIRECTORY HAS stack-env.sh THEN INCLUDE IT
+[ -e ./stack-env.sh ] && . ./stack-env.sh
+
+if [ -z "$ENVIRONMENT" ]; then
+   echo "No ENVIRONMENT provided or found.  Exiting ..."
+   exit 201
+fi
+
+[ -e ./sites/$ENVIRONMENT/stack-env.sh ] && . ./sites/$ENVIRONMENT/stack-env.sh
+
+LOCAL_PATH=$(dirname "${BASH_SOURCE[0]}")
+
+[ -z "$ROLE" ] && ROLE="nomad-pool"
+[ -z "$POOL_TYPE" ] && POOL_TYPE="general"
+[ -z "$NAME" ] && NAME="$ENVIRONMENT-$ORACLE_REGION-$ROLE-$POOL_TYPE"
+[ -z "$ORACLE_GIT_BRANCH" ] && ORACLE_GIT_BRANCH="main"
+
+[ -z "$POOL_PUBLIC" ] && POOL_PUBLIC="false"
+
+[ -e "$LOCAL_PATH/../../clouds/all.sh" ] && . $LOCAL_PATH/../../clouds/all.sh
+[ -e "$LOCAL_PATH/../../clouds/oracle.sh" ] && . $LOCAL_PATH/../../clouds/oracle.sh
+
+if [ -z "$ORACLE_REGION" ]; then
+  echo "No ORACLE_REGION found. Exiting..."
+  exit 203
+fi
+
+if [ -z "$INFRA_CONFIGURATION_REPO" ]; then
+  echo "No INFRA_CONFIGURATION_REPO found. Exiting..."
+  exit 203
+fi
+
+if [ -z "$INFRA_CUSTOMIZATIONS_REPO" ]; then
+  echo "No INFRA_CUSTOMIZATIONS_REPO found. Exiting..."
+  exit 203
+fi
+
+ORACLE_CLOUD_NAME="$ORACLE_REGION-$ENVIRONMENT-oracle"
+[ -e "$LOCAL_PATH/../../clouds/${ORACLE_CLOUD_NAME}.sh" ] && . $LOCAL_PATH/../../clouds/${ORACLE_CLOUD_NAME}.sh
+
+[ -z "$SHAPE" ] && SHAPE="$DEFAULT_NOMAD_POOL_SHAPE"
+[ -z "$SHAPE" ] && SHAPE="$SHAPE_A_1"
+
+[ -z "$MEMORY_IN_GBS" ] && MEMORY_IN_GBS="64"
+if [[ "$SHAPE" == "VM.Standard.A1.Flex" ]]; then
+  [ -z "$OCPUS" ] && OCPUS="12"
+fi
+
+[ -z "$OCPUS" ] && OCPUS="6"
+[ -z "$DISK_IN_GBS" ] && DISK_IN_GBS="100"
+
+[ -z "$INSTANCE_POOL_SIZE" ] && INSTANCE_POOL_SIZE=1
+
+[ -z "$NAME_ROOT" ] && NAME_ROOT="$NAME"
+
+[ -z "$INSTANCE_POOL_NAME" ] && INSTANCE_POOL_NAME="${NAME_ROOT}-InstancePool"
+[ -z "$INSTANCE_CONFIG_NAME" ] && INSTANCE_CONFIG_NAME="${NAME_ROOT}-InstanceConfig"
+
+RESOURCE_NAME_ROOT="${NAME_ROOT}"
+
+[ -z "$POSTRUNNER_PATH" ] && export POSTRUNNER_PATH="terraform/nomad-whisper/user-data/postinstall-runner-nomad-whisper-oracle.sh"
+
+if [[ "$POOL_PUBLIC" == "true" ]]; then
+  POOL_SUBNET_OCID="$PUBLIC_SUBNET_OCID"
+  EPHEMERAL_INGRESS_CIDR="0.0.0.0/0"
+else
+  POOL_SUBNET_OCID="$NAT_SUBNET_OCID"
+  EPHEMERAL_INGRESS_CIDR="10.0.0.0/8"
+fi
+
+[ -z "$ENCRYPTED_CREDENTIALS_FILE" ] && ENCRYPTED_CREDENTIALS_FILE="$LOCAL_PATH/../../ansible/secrets/ssl-certificates.yml"
+[ -z "$VAULT_PASSWORD_FILE" ] && VAULT_PASSWORD_FILE="$LOCAL_PATH/../../.vault-password.txt"
+
+# look up instance pool. If it exists, find its curent size and set INSTANCE_POOL_SIZE appropriately
+INSTANCE_POOL_DETAILS="$(oci compute-management instance-pool list --region "$ORACLE_REGION" -c "$COMPARTMENT_OCID" --all --display-name "$INSTANCE_POOL_NAME" | jq '.data[0]')"
```

Comment:
We have no instance pool, so instead we need to look up the equivalent autoscale group